DQM/SiPixelPhase1RecHits/src/SiPixelPhase1RecHits.cc

```diff
@@ -30,24 +37,60 @@ void SiPixelPhase1RecHits::analyze(const edm::Event& iEvent, const edm::EventSet
   iSetup.get<TrackerDigiGeometryRecord>().get(tracker);
   assert(tracker.isValid());
 
-  edm::Handle<SiPixelRecHitCollection> input;
-  iEvent.getByToken(srcToken_, input);
-  if (!input.isValid()) return;
+  edm::Handle<reco::TrackCollection> tracks;
+  iEvent.getByToken( srcToken_, tracks);
+  if (!tracks.isValid()) return;
+
+  for (auto const & track : *tracks) {
 
-  SiPixelRecHitCollection::const_iterator it;
-  for (it = input->begin(); it != input->end(); ++it) {
-    auto id = DetId(it->detId());
+    bool isBpixtrack = false, isFpixtrack = false;
 
-    for(SiPixelRecHit const& rechit : *it) {
-      SiPixelRecHit::ClusterRef const& clust = rechit.cluster();
+    auto const & trajParams = track.extra()->trajParams();
+    auto hb = track.recHitsBegin();
+    for(unsigned int h=0;h<track.recHitsSize();h++){
+     
+      auto hit = *(hb+h);
```

Comment:
For the (near) future - it would be better to avoid this copy - perhaps it is not possible with the current track interface, but should be.